cpu/atmega_common/periph/gpio.c

```diff
@@ -148,43 +148,26 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
     /* clear global interrupt flag */
     cli();
 
+#if defined(CPU_ATMEGA328P)
+    /* INT pins start at PD2 instead of at PD0 */
+    pin_num -= 2;
+#endif
+
     EIMSK |= (1 << pin_num);
 
     /* configure the flank */
-    switch (flank) {
-        case GPIO_RISING:
-            if (pin_num < 4) {
-                EICRA |= (3 << pin_num * 2);
-            }
-#if defined(EICRB)
-            else {
-                EICRB |= (3 << (pin_num * 2) % 4);
-            }
-#endif
-            break;
-        case GPIO_FALLING:
-            if (pin_num < 4) {
-                EICRA |= (2 << pin_num * 2);
-            }
-#if defined(EICRB)
-            else {
-                EICRB |= (2 << (pin_num * 2) % 4);
-            }
-#endif
-            break;
-        case GPIO_BOTH:
-            if (pin_num < 4) {
-                EICRA |= (1 << pin_num * 2);
-            }
+    if (flank > 3) {
```

Comment:
`3` looks like a magic number here, should be `GPIO_FLANK_RISING` or alternatively define `GPIO_FLANK_MAX` and check `(flank < GPIO_FLANK_MAX)`.

Reply:
Guilty :-P. Will fix :+1: